designer/client/conditions/AbsoluteDateValues.tsx

```diff
@@ -19,76 +20,125 @@ interface Props {
   updateValue: ({ year, month, day }: YearMonthDay) => void;
 }
 
+function isValidateDate(props: {
+  year?: number | string;
+  month?: number | string;
+  day?: number | string;
+}) {
+  const { year = 2020, month = 12, day = 1 } = props;
+  const parsedDate = Date.parse(`${year}-${month}-${day}`);
+
+  if (isNaN(parsedDate)) {
+    return false;
+  }
+
+  return true;
+}
+
+function isValidDay(day: string) {
+  return isValidateDate({ day });
+}
+function isValidMonth(month: string) {
+  return isValidateDate({ month });
+}
+function isValidYear(year: string) {
+  return year.length === 4 && isValidateDate({ year });
+}
+
 export const AbsoluteDateValues = ({ value = {}, updateValue }: Props) => {
-  const [year, setYear] = React.useState<string>(() =>
+  const [year, setYear] = useState<string>(() =>
     isInt(value.year) ? (value.year as number).toString() : ""
   );
-  const [month, setMonth] = React.useState<string>(() =>
-    isInt(value.month) ? (value.month as number).toString() : ""
+
+  const [month, setMonth] = useState<string>(() =>
+    isInt(value.month) ? padStart(String(value.month), 2, "0") : ""
   );
-  const [day, setDay] = React.useState<string>(() =>
-    isInt(value.day) ? (value.day as number).toString() : ""
+
+  const [day, setDay] = useState<string>(() =>
+    isInt(value.day) ? padStart(String(value.day), 2, "0") : ""
   );
 
-  React.useEffect(() => {
-    const parsedYear = tryParseInt(year);
-    const parsedMonth = tryParseInt(month);
-    const parsedDay = tryParseInt(day);
+  useEffect(() => {
+    const parsedDay = tryParseInt(day)!;
+    const parsedMonth = tryParseInt(month)!;
+    const parsedYear = tryParseInt(year)!;
+
     if (
-      parsedYear &&
-      parsedMonth &&
-      parsedDay &&
-      (parsedYear !== value.year ||
-        parsedMonth !== value.month ||
-        parsedDay !== value.day) &&
-      isValid(new Date(parsedYear, parsedMonth - 1, parsedDay))
+      parsedDay === value.day &&
+      parsedMonth === value.month &&
+      parsedYear === value.year
     ) {
+      return;
+    }
+
+    if (!isValidDay(day) || !isValidMonth(month) || !isValidYear(year)) {
+      return;
+    }
+
+    if (isValid(new Date(parsedYear, parsedMonth - 1, parsedDay))) {
       updateValue({ year: parsedYear, month: parsedMonth, day: parsedDay });
     }
   }, [year, month, day]);
 
-  const yearChanged = (e) => setYear(e.target.value);
-  const monthChanged = (e) => setMonth(e.target.value);
-  const dayChanged = (e) => setDay(e.target.value);
+  const dayChanged = (e: ChangeEvent<HTMLInputElement>) => {
+    const day = e.target.value;
+    if (Number(day) <= 31) {
```

Comment:
Do you need this, if the number input has a max value of 31?

Reply:
QA was able to break it, and manually testing it I am also able to insert any number in days and months using Chrome, max prop is being totally ignored.

Reply:
👍